### PR TITLE
[flang] Add FIR AliasAnalysis alias() wrapper to allow external getSource() method

### PR DESCRIPTION
**Pull request by SusanTan**
Adding a wrapper around alias(mlir::Value lhs, mlir::Value rhs) to allow user to provide Source objects.

### CODE REVIEW

File: flang/lib/Optimizer/Analysis/AliasAnalysis.cpp [hunk 1]
```diff
@@ -12,6 +12,7 @@
 #include "flang/Optimizer/Dialect/FIRType.h"
 #include "flang/Optimizer/Dialect/FortranVariableInterface.h"
 #include "flang/Optimizer/HLFIR/HLFIROps.h"
+#include "flang/Optimizer/CodeGen/CGOps.h"
```
Comment (razvanlupusoru) [inline flang/lib/Optimizer/Analysis/AliasAnalysis.cpp:15]: This include is not needed.

File: flang/lib/Optimizer/Analysis/AliasAnalysis.cpp [hunk 2]
```diff
@@ -130,18 +131,25 @@ bool AliasAnalysis::Source::mayBeActualArgWithPtr(
...
   // not recognize non-aliasing entities.
-  auto lhsSrc = getSource(lhs);
-  auto rhsSrc = getSource(rhs);
   bool approximateSource = lhsSrc.approximateSource || rhsSrc.approximateSource;
   LLVM_DEBUG(llvm::dbgs() << "\nAliasAnalysis::alias\n";
              llvm::dbgs() << "  lhs: " << lhs << "\n";
              llvm::dbgs() << "  lhsSrc: " << lhsSrc << "\n";
              llvm::dbgs() << "  rhs: " << rhs << "\n";
              llvm::dbgs() << "  rhsSrc: " << rhsSrc << "\n";);
-
```
Comment (razvanlupusoru) [inline flang/lib/Optimizer/Analysis/AliasAnalysis.cpp:144]: Extra line should be OK to stay.

**Comment by razvanlupusoru** [comment]
Looks great to me to make this improvement. By separating in this way, I can imagine how it would be useful for handling acc dialect operations (which FIR's AA cannot currently handle). Thank you.

I would appreciate @Renaud-K look at it and give his feedback.

**Review by Renaud-K — APPROVED** [review]
This looks great. Strangely when I look at the changes from all commits, I am not seeing the new CG Op change. I am wondering if something did go wrong on a rebase or a push. You can always push the change again if it is the case.

**Comment by SusanTan** [comment]
> This looks great. Strangely when I look at the changes from all commits, I am not seeing the new CG Op change. I am wondering if something did go wrong on a rebase or a push. You can always push the change again if it is the case.

I removed the CG changes because that was causing some test cases failing, and it's not necessary for downstream work.

**Review by clementval — APPROVED** [review]
LGTM. note: If you use `git-clang-format`, it will only format the line you touch in your change so you don't have unrelated change in your PR.

**Review by razvanlupusoru — APPROVED** [review]

**Pull request merged**